commands/proposal_submission.go

```diff
@@ -566,23 +566,34 @@ func checkDataSourceSpec(spec *vegapb.DataSourceDefinition, name string, parentP
 		return errs.FinalAddForProperty(fmt.Sprintf("%s.%s", parentProperty, name), ErrIsRequired)
 	}
 
-	signers := spec.GetSigners()
-	filters := spec.GetFilters()
-	if isBuiltInSpec(filters) {
-		return checkDataSourceSpecFilters(filters, name, parentProperty)
+	if spec.SourceType == nil {
+		return errs.FinalAddForProperty(fmt.Sprintf("%s.%s", parentProperty, name+".source_type"), ErrIsRequired)
 	}
 
-	if len(signers) == 0 {
-		errs.AddForProperty(fmt.Sprintf("%s.%s.signers", parentProperty, name), ErrIsRequired)
-	}
-	for i, key := range signers {
-		signer := types.SignerFromProto(key)
-		if signer.IsEmpty() {
-			errs.AddForProperty(fmt.Sprintf("%s.%s.signers.%d", parentProperty, name, i), ErrIsNotValid)
+	switch tp := spec.SourceType.(type) {
+	case *vegapb.DataSourceDefinition_External:
+		// For now check only for oracle type for external data source. Add a check for Oracle type data later when other sources are added.
+		o := tp.External.GetOracle()
+
+		signers := o.Signers
+		if len(signers) == 0 {
+			errs.AddForProperty(fmt.Sprintf("%s.%s.external.oracle.signers", parentProperty, name), ErrIsRequired)
+		}
+
+		for i, key := range signers {
+			signer := types.SignerFromProto(key)
+			if signer.IsEmpty() {
+				errs.AddForProperty(fmt.Sprintf("%s.%s.external.oracle.signers.%d", parentProperty, name, i), ErrIsNotValid)
+			}
 		}
-	}
 
-	errs.Merge(checkDataSourceSpecFilters(filters, name, parentProperty))
+		filters := o.Filters
+		if isBuiltInSpec(filters) {
+			return checkDataSourceSpecFilters(filters, fmt.Sprintf("external.oracle.%s", name), parentProperty)
+		}
+
+		errs.Merge(checkDataSourceSpecFilters(filters, fmt.Sprintf("external.oracle.%s", name), parentProperty))
+	}
```

Comment:
This is wrong, it should be ```fmt.Sprintf("%s.external.oracle", name)```
I posted it wrong, let me fix it.

Reply:
should it be %s.external.oracle? that would be e.g. `signers.external.oracle` vs `external.oracle.signers` as it is now.

Reply:
ah no, it's moving the dataspec backwards, no worries, I'll fix.

Reply:
yes it should be data_spec_for_settlement_data.external.oracle.signers/filters